core/lib/spree/core/controller_helpers/order.rb

```diff
@@ -37,6 +37,12 @@ def current_order(options = {})
         end
 
         def associate_user
+          return if Spree::Config.associate_user_in_authentication_extension
+          Spree::Deprecation.behavior = :stderr
+          Spree::Deprecation.warn(
+            "Order#associate_user is deprecated and will be removed in 3.2.0.alpha",
```

Comment:
The deprecation horizon should be beyond `3.2`, as that version will be the one adding the new option. I think our policy is not breaking code until a major release, so it should be `4.0.0`. cc @kennyadsl

Reply:
You are right @waiting-for-dev, the method will be removed in the next major (`4.0.0`).